addons/api/addon/serializers/credential-library.js

```diff
@@ -16,6 +16,10 @@ export default class CredentialLibrarySerializer extends ApplicationSerializer {
   serialize() {
     const serialized = super.serialize(...arguments);
     if (serialized.attributes) {
+      if (serialized.type !== 'vault-generic') {
+        delete serialized.attributes.http_method;
+      }
+
       // Serialize `http_request_body` only if `http_method` is POST
       if (!serialized.attributes?.http_method?.match(/post/i))
         delete serialized.attributes.http_request_body;
```

Comment:
Is there a clearer / more direct way of serializing fields by type here?

Reply:
Hmm I thought about it and wasn't sure what pattern we should do here. I actually didn't want to do anything with the serializer here but the issue I encountered was we were providing a default value for the `http_method` on the `vault-generic`. 

I couldn't make the default dependent based on the type (do you know if we can?) and setting the default from the `vault-generic` form was ugly as I would have to manage both a local variable and the model state for it.